### DISCUSSION TITLE
Hey ! I think I have the same issue and none of the answers has worked.

### DISCUSSION THREAD

**Discussion by AlexandreLarribau** [micropython/micropython#11024] [General]
Hey ! I think I have the same issue and none of the answers has worked. 

I am listenning 5 pins. And I try to guess which has raised the interrupt. 

p1 = machine.Pin(1, machine.Pin.IN)
p2 = machine.Pin(2, machine.Pin.IN)
p4 = machine.Pin(4, machine.Pin.IN)
p5 = machine.Pin(5, machine.Pin.IN)
p6 = machine.Pin(6, machine.Pin.IN)

def handle_pwm_interrupt(pin):
  
    print("the pin is : ")
    print(pin)

p1.irq(trigger=machine.Pin.IRQ_FALLING | machine.Pin.IRQ_RISING, handler=handle_pwm_interrupt)
p2.irq(trigger=machine.Pin.IRQ_FALLING | machine.Pin.IRQ_RISING, handler=handle_pwm_interrupt)
p4.irq(trigger=machine.Pin.IRQ_FALLING | machine.Pin.IRQ_RISING, handler=handle_pwm_interrupt)
p5.irq(trigger=machine.Pin.IRQ_FALLING | machine.Pin.IRQ_RISING, handler=handle_pwm_interrupt)
p6.irq(trigger=machine.Pin.IRQ_FALLING | machine.Pin.IRQ_RISING, handler=handle_pwm_interrupt)

when I do this the display is "Pin(GPIO4, mode=IN)". 
I tried pin.name(), pin.id(), pin.id and each time i get :"File "<stdin>", line 16, in handle_pwm_interrupt
AttributeError: 'Pin' object has no attribute 'name'" 

And with id(pin) i get "268703464" which doesn't help me much to know which pin it is. 

Do you have advices ?

_Originally posted by @AlexandreLarribau in https://github.com/micropython/micropython/issues/2612#issuecomment-1466308346_
            
![Capture](https://user-images.githubusercontent.com/72812521/224743260-72ec8dbd-b792-4551-97be-b27b2ef52b96.PNG)

**Comment by rkompass** [comment]
You have the pin object in the interrupt handler. So you should be able to compare like:
```
if pin == p1:
    do_something1()
elif pin == p2:
    do_something2()
elif pin == p3:
    do_something3()
... 
```
and so on. Getting the pin number from the pin object is a known cumbersomeness in Micropython: You have to extract it from the string representation. Like:
`gpio_id = int(str(pin).split(", ")[0][4:])` which worked on the RPi Pico.
For the RP2 I found another way, that is not simpler though, iirc.

I see now I repeated old answers/advices.

In the headings you state: **None of the answers has worked.**

Please elaborate what you tried and how it didn't work.

Perhaps you wanted to urge on this problem (is this correct English?). If you wanted to start a query on that feature (having a pin.id() function) I would support it. But consider: Pins are sometimes numbers, sometimes letters, ...

**Comment by robert-hh** [comment]
I'll convert that to a discussion, which is the better place to deal with it.

**Comment by robert-hh** [comment]
Ok.- So i hooke it up to a board. Two problems with that sample script came along:
- You cannot use Pin(1), because that is TX of the UART. If you define that as Input, you loose the connection to REPL
- Pin(6) does not exist.

Besides that, here a sample script with one of the solutions:
```
import machine

p2 = machine.Pin(2, machine.Pin.IN)
p4 = machine.Pin(4, machine.Pin.IN)
p5 = machine.Pin(5, machine.Pin.IN)

def handle_pwm_interrupt(pin):
    global p2, p4, p5, p6
    if pin is p2:
         print("the pin is : p2")
    elif pin is p4:
         print("the pin is : p4")
    elif pin is p5:
         print("the pin is : p5")

p2.irq(trigger=machine.Pin.IRQ_FALLING , handler=handle_pwm_interrupt)
p4.irq(trigger=machine.Pin.IRQ_FALLING , handler=handle_pwm_interrupt)
p5.irq(trigger=machine.Pin.IRQ_FALLING , handler=handle_pwm_interrupt)
```
Tested with a Wemos D1 Mini and a signal generator. I do not know your application and what you want to do with the pin, but the basic mechanism works. Alternatively you could define a separate callback for each pin, and call a common handler by these.
